### PR TITLE
MDS-3848 - init build refactor for filesystem-provider

### PR DESCRIPTION
**Pull request by wilson496**
# Main

Refactoring build process for filesystem-provider service.

# Other
- N/A

# How to test

- Tag and build through oc tag command (ensure you're logged into the OC console)
- Check if the container is running
- Boom, done. 

# Notes

- N/A

### CODE REVIEW

File: services/filesystem-provider/Dockerfile.ci [hunk 1]
```diff
@@ -0,0 +1,21 @@
+FROM mcr.microsoft.com/dotnet/sdk:3.1
```
Comment (jon-funk) [inline services/filesystem-provider/Dockerfile.ci:1]: Just a note, not requesting change: As per the norm, if the resulting build doesn't work you may need to look at https://github.com/bcgov/mds/blob/develop/openshift4/docker-images/dotnet-31/Dockerfile as that's what we had to use for permissions to get it to work in the past, however that is a redhat image.

File: .github/workflows/build-filesystem-provider.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      - develop
+    paths:
+      - migrations/**
+      - .github/workflows/build-filesystem-provider.yaml
+
+env:
+  INITIAL_TAG: latest
+  TAG: dev
+  NAME: filesystem-provider
+  CONTEXT: migrations/
```
Comment (jon-funk) [inline .github/workflows/build-filesystem-provider.yaml]: Update context path

File: .github/workflows/build-filesystem-provider.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
+name: Build filesystem-provider
+
+on:
+  pull_request:
+  push:
+    branches:
+      - develop
+    paths:
+      - migrations/**
```
Comment (jon-funk) [inline .github/workflows/build-filesystem-provider.yaml]: update path trigger to match where devs would be making changes

Comment (wilson496) [reply .github/workflows/build-filesystem-provider.yaml]: fixed

**Review by jon-funk — APPROVED** [review]
Great work!!

**Pull request merged**